agent/installer/installer.go

```diff
@@ -85,6 +91,53 @@ func (bd *bundleDownloader) DownloadOrPreview(os, k8s, tag string) error {
 	return bd.Download(os, k8s, tag)
 }
 
+func supportedOSversion(os string) error {
+	matched, err := regexp.Match(SupportedOS, []byte(os))
+	if err != nil {
+		return err
+	}
+	if !matched {
+		errMsg := fmt.Sprintf("Unsupported OS. Note: Currently supported OS is %s", SupportedOS)
+		return errors.New(errMsg)
+	}
+	return nil
+}
+
+func ckeckPreRequsitePackages() error {
+	unavailablePackages := []string{}
+	for _, pkgName := range PreRequisitePackages {
+		_, err := exec.Command("dpkg-query", "-W", pkgName).Output()
```

Comment:
cc @georgievaVMW

Reply:
The code is OS/Platform agnostic, except for this line I guess. We need a better way to do this check. We can either use the current installer approach and provide a matrix of <package type> to <package manager cmd> mapping or provide the package type information externally. Both have pros and cons. Any proposals are very welcome.

Reply:
I am slightly hesitant, since this feels like a duplicate of the kubeadm preflight checks. But I understand that we don't have kubeadm when the host-agent starts, and if we wait until we have kubeadm - its too late.

Can we either reuse what kubeadm does? Or maybe follow inspiration from them? https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L1032-L1043

Reply:
Wow, this was very helpful.
Fixed.